android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/AndroidPayCheckoutSession.java

```diff
@@ -0,0 +1,128 @@
+package com.shopify.unity.buy;
+
+import com.shopify.buy3.pay.PayCart;
+import com.shopify.unity.buy.utils.AndroidLogger;
+import com.shopify.unity.buy.utils.ILogger;
+import com.unity3d.player.UnityPlayer;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.math.BigDecimal;
+
+public final class AndroidPayCheckoutSession {
+    private ILogger logger;
+
+    public AndroidPayCheckoutSession() {
+        this.logger = new AndroidLogger();
+    }
+
+    public AndroidPayCheckoutSession(ILogger logger) {
+        this.logger = logger;
+    }
+
+    public boolean checkoutWithAndroidPay(
+            String unityDelegateObjectName,
+            String merchantName,
+            String summaryItemsString,
+            String currencyCode,
+            String countryCode,
+            boolean requiresShipping
+    ) {
+        try {
+            PayCart cart = cartFromUnity(merchantName, summaryItemsString, currencyCode, countryCode,
+                    requiresShipping);
+            addAndroidPayFragment(unityDelegateObjectName, cart, countryCode);
+            return true;
+        } catch (JSONException e) {
+            logger.error("ShopifyBuyPlugin", "Failed to parse summary items from Unity");
+            e.printStackTrace();
+            return false;
+        } catch (IllegalArgumentException e) {
+            logger.error("ShopifyBuyPlugin", e.getMessage());
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public PayCart cartFromUnity(
+        String merchantName,
+        String summaryItemsString,
+        String currencyCode,
+        String countryCode,
```

Comment:
Pls add TODO to fix it later with new version of `PayCart`. I'm going to add this and make a new release of pay support